src/components/Collection/CollectionDetail/CollectionDetail.tsx

```diff
@@ -1,51 +1,178 @@
+import { useRouter } from 'next/router';
+import { useSelector, shallowEqual } from 'react-redux';
+
+import ChevronDownIcon from '../../../../public/icons/chevron-down.svg';
+import OverflowMenuIcon from '../../../../public/icons/menu_more_horiz.svg';
+import CollectionSorter from '../CollectionSorter/CollectionSorter';
 
 import styles from './CollectionDetail.module.scss';
 
-import Collapsible from '@/dls/Collapsible/Collapsible';
-import ChevronDownIcon from '@/icons/chevron-down.svg';
-import OverflowMenuIcon from '@/icons/menu_more_horiz.svg';
-
-type CollectionItem = {
-  id: string | number;
-  title: string;
-  content: string;
-};
+import { logButtonClick } from '@/utils/eventLogger';
+import { getChapterWithStartingVerseUrl } from '@/utils/navigation';
+import DataFetcher from 'src/components/DataFetcher';
+import Button, { ButtonVariant } from 'src/components/dls/Button/Button';
+import Collapsible from 'src/components/dls/Collapsible/Collapsible';
+import PopoverMenu from 'src/components/dls/PopoverMenu/PopoverMenu';
+import TafsirVerseText from 'src/components/QuranReader/TafsirView/TafsirVerseText';
+import TranslationText from 'src/components/QuranReader/TranslationView/TranslationText';
+import DataContext from 'src/contexts/DataContext';
+import { selectQuranReaderStyles } from 'src/redux/slices/QuranReader/styles';
+import { selectSelectedTranslations } from 'src/redux/slices/QuranReader/translations';
+import { getDefaultWordFields, getMushafId } from 'src/utils/api';
+import { makeVersesUrl } from 'src/utils/apiPaths';
+import { areArraysEqual } from 'src/utils/array';
+import { deleteCollectionBookmark } from 'src/utils/auth/api';
+import { getChapterData } from 'src/utils/chapter';
+import { toLocalizedVerseKey } from 'src/utils/locale';
+import { makeVerseKey } from 'src/utils/verse';
+import { VersesResponse } from 'types/ApiResponses';
+import Bookmark from 'types/Bookmark';
 
 type CollectionDetailProps = {
+  id: string;
   title: string;
-  collectionItems: CollectionItem[];
+  bookmarks: Bookmark[];
+  sortBy: string;
+  onSortByChange: (sortBy: string) => void;
+  onUpdated: () => void;
 };
 
-const CollectionDetail = ({ title, collectionItems }: CollectionDetailProps) => {
-  const { t } = useTranslation();
+enum CollectionDetailSortOption {
+  RecentlyAdded = 'recentlyAdded',
+  VerseKey = 'verseKey',
+}
+
+const CollectionDetail = ({
+  id,
+  title,
+  bookmarks,
+  sortBy,
+  onSortByChange,
+  onUpdated,
+}: CollectionDetailProps) => {
+  const { t, lang } = useTranslation();
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const { quranFont, mushafLines } = quranReaderStyles;
+  const { mushaf } = getMushafId(quranFont, mushafLines);
+  const selectedTranslations = useSelector(selectSelectedTranslations, areArraysEqual);
+
+  const onBookmarkItemDeleted = (bookmarkId: string) => {
+    deleteCollectionBookmark(id, bookmarkId).then(() => {
+      onUpdated();
+    });
```

Comment:
What happens when there is an issue with the API? I think we should show a toast/error message